firedrake/interpolation.py

```diff
@@ -215,8 +250,68 @@ def __init__(
         self._allow_missing_dofs = allow_missing_dofs
         self.callable = None
 
+    def _interpolate_future(self, *function, transpose, default_missing_val):
+
+        V = self.V
+        if isinstance(V, firedrake.Function):
+            V = V.function_space()
+
+        interp_data = {'subset': self.subset, 'freeze_expr': self.freeze_expr,
+                       'access': self.access, 'bcs': self.bcs,
+                       'allow_missing_dofs': self._allow_missing_dofs,
+                       'default_missing_val': default_missing_val}
+        interp = Interpolate(self.expr, V, interp_data=interp_data)
+        if transpose:
+            interp = adjoint(interp)
+
+        if function:
+            f, = function
+            # Passing in a function is equivalent to taking the action.
+            interp = action(interp, f)
+        # Return the `ufl.Interpolate` object
+        return interp
+
+    @PETSc.Log.EventDecorator()
+    def interpolate(self, *function, output=None, transpose=False, default_missing_val=None):
+        """This performs interpolation by assembling `Interp`, which in turn calls
```

Comment:
Docstrings should be imperative (e.g. `"Interpolate an expression."`) and short enough to fit on a line. Leave a blank line between this and the rest of the information.

Reply:
And should this be `Interpolate` instead of `Interp`?